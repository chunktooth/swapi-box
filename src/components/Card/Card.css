.Card {
	width: 30%;
	background: hsla(0, 0%, 0%, .7);
	color: #92ACA3;
	border: 2px solid #2B3536;
	border-radius: 25px;
	margin: 10px;
	padding: 7px;
	box-shadow: 0px 0px 10px 2px #0ff;
}

.Card-header {
	padding: 3px;
	color: #F3DA4A;
	font-family: 'Quantico', sans-serif;
}

.favorite {
	background-color: transparent;
	border: none;
}

.diskette {
	width: 26px;
	margin-bottom: 16px;
}

.diskette:hover {
	box-shadow: 0px 0px 20px 3px #F3DA4A;
}