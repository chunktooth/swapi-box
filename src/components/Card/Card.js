import React from 'react';
import diskette from '../../images/diskette.svg';
import PropTypes from 'prop-types';
import './Card.css';

const Card = ({data, setFavorites}) => {

  const cardKeys = Object.keys(data);
  const filteredKeys = cardKeys.filter(key => key !== 'name');
  const selectedData = filteredKeys.map((selectedKey, index) => {
    const capitalizedKey = selectedKey.charAt(0).toUpperCase() + selectedKey.substr(1);

   return <p key={index}>
      <span className="keyName">{capitalizedKey}: </span>
      {data[selectedKey]}
      </p>
  });

  return (
    <article className="Card">
      <h3 className="Card-header">{data.name}</h3>
      <div>{selectedData}</div>
      <button 
        className="favorite"
        onClick={() => setFavorites(data)}>
        <img 
          src={diskette}
          className="diskette"
          alt='Save as favorites'/>
      </button>
    </article>
  );
};

Card.propTypes = {
  data: PropTypes.object.isRequired,
  setFavorites: PropTypes.func
}

export default Card;