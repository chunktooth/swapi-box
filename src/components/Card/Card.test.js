import React from 'react';
import Card from './Card';
import {Shallow} from 'enzyme';

describe('Card', () => {
	let card;

	beforeEach(() => {
		card = shallow(<Card />);
	});

	it('should match the snapshot', () => {
		expect(card).toMatchSnapshot();
	});
	
})