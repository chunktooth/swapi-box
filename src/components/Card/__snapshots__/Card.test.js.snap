// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "ears": "long",
            "name": "Jar-jar",
            "status": "questionable",
          }
    }
    setFavorites={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="Card-header"
>
          Jar-jar
</h3>,
        <div>
          <p>
                    <span
                              className="keyName"
                    >
                              Ears
                              : 
                    </span>
                    long
          </p>
          <p>
                    <span
                              className="keyName"
                    >
                              Status
                              : 
                    </span>
                    questionable
          </p>
</div>,
        <button
          className="favorite"
          onClick={[Function]}
>
          <img
                    alt="Save as favorites"
                    className="diskette"
                    src="diskette.svg"
          />
</button>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Jar-jar",
          "className": "Card-header",
        },
        "ref": null,
        "rendered": "Jar-jar",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              <span
                            className="keyName"
              >
                            Ears
                            : 
              </span>
              long
</p>,
            <p>
              <span
                            className="keyName"
              >
                            Status
                            : 
              </span>
              questionable
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="keyName"
>
                  Ears
                  : 
</span>,
                "long",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Ears",
                    ": ",
                  ],
                  "className": "keyName",
                },
                "ref": null,
                "rendered": Array [
                  "Ears",
                  ": ",
                ],
                "type": "span",
              },
              "long",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="keyName"
>
                  Status
                  : 
</span>,
                "questionable",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Status",
                    ": ",
                  ],
                  "className": "keyName",
                },
                "ref": null,
                "rendered": Array [
                  "Status",
                  ": ",
                ],
                "type": "span",
              },
              "questionable",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Save as favorites"
            className="diskette"
            src="diskette.svg"
/>,
          "className": "favorite",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Save as favorites",
            "className": "diskette",
            "src": "diskette.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "button",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="Card-header"
>
            Jar-jar
</h3>,
          <div>
            <p>
                        <span
                                    className="keyName"
                        >
                                    Ears
                                    : 
                        </span>
                        long
            </p>
            <p>
                        <span
                                    className="keyName"
                        >
                                    Status
                                    : 
                        </span>
                        questionable
            </p>
</div>,
          <button
            className="favorite"
            onClick={[Function]}
>
            <img
                        alt="Save as favorites"
                        className="diskette"
                        src="diskette.svg"
            />
</button>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Jar-jar",
            "className": "Card-header",
          },
          "ref": null,
          "rendered": "Jar-jar",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                <span
                                className="keyName"
                >
                                Ears
                                : 
                </span>
                long
</p>,
              <p>
                <span
                                className="keyName"
                >
                                Status
                                : 
                </span>
                questionable
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="keyName"
>
                    Ears
                    : 
</span>,
                  "long",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Ears",
                      ": ",
                    ],
                    "className": "keyName",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Ears",
                    ": ",
                  ],
                  "type": "span",
                },
                "long",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="keyName"
>
                    Status
                    : 
</span>,
                  "questionable",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Status",
                      ": ",
                    ],
                    "className": "keyName",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Status",
                    ": ",
                  ],
                  "type": "span",
                },
                "questionable",
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Save as favorites"
              className="diskette"
              src="diskette.svg"
/>,
            "className": "favorite",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Save as favorites",
              "className": "diskette",
              "src": "diskette.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "button",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
